### PR TITLE
Feature protected routes

### PR DESCRIPTION
**Pull request by mayanksuchde**
### What this PR does (required):
- Added a protected route HOC which checks if user is logged in or not,if yes sends them to where they want ot go , if not redirects to login
- 

### Screenshots / Videos (required):
- When I try to access dashboard without logging in:
Screen Shot 2021-09-23 at 1 16 32 PM
- It redirects to login
Screen Shot 2021-09-23 at 1 17 02 PM

- When logged in:it send me to dashboard as it is suppose to
Screen Shot 2021-09-23 at 1 17 25 PM

### Any information needed to test this feature (required):
- have removed to the 'history.push(/login)' from dasboard to see if ProtectedRoute works
- 
### Any issues with the current functionality (optional):
- None

### CODE REVIEW

**Review by ezeilo-su — COMMENTED** [review]
Great you added this feature.
Happy coding!

File: client/src/components/ProtectedRoute/ProtectedRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+const ProtectedRoute = ({ children, ...rest }: ProtectedRouteProps): JSX.Element => {
+  const { loggedInUser } = useAuth();
+
+  return (
+    <>
+      <Route
+        {...rest}
+        render={({ location }) =>
+          loggedInUser ? children : <Redirect to={{ pathname: '/login', state: { from: location } }} />
```
Comment (ugur-akin) [inline client/src/components/ProtectedRoute/ProtectedRoute.tsx:18]: This is nice to see 🙌🏼

**Review by ugur-akin — COMMENTED** [review]
LGTM! 👍🏼

**Review by rajivtitus — APPROVED** [review]
LGTM! :)

**Pull request merged**